test/functional/wallet_avoidreuse.py

```diff
@@ -257,5 +258,35 @@ def test_fund_send_fund_send(self, second_addr_type):
         assert_approx(self.nodes[1].getbalance(), 1, 0.001)
         assert_approx(self.nodes[1].getbalance(avoid_reuse=False), 11, 0.001)
 
+    def test_getbalances_used(self):
+        '''
+        getbalances and listunspent should pick up on reused addresses
+        immediately, even for address reusing outputs created before the first
+        transaction was spending from that address
+        '''
+        self.log.info("Test getbalances used category")
+
+        # node under test should be completely empty
+        assert_equal(self.nodes[1].getbalance(avoid_reuse=False), 0)
+
+        new_addr = self.nodes[1].getnewaddress()
+        ret_addr = self.nodes[0].getnewaddress()
+
+        # send multiple transactions, reusing one address
+        for _ in range(11):
+            self.nodes[0].sendtoaddress(new_addr, 1)
+
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        # send transaction that should not use all the available outputs
+        # per the current coin selection algorithm
+        self.nodes[1].sendtoaddress(ret_addr, 5)
+
+        # getbalances and listunspent should show the remaining outputs
+        # in the reused address as used/reused
+        assert_unspent(self.nodes[1], total_count=2, total_sum=6, reused_count=1, reused_sum=1)
+        assert_balances(self.nodes[1], mine={"used": 1, "trusted": 5})
```

Comment:
Why is reused sum 1 and not 6? All of the UTXOs should be counted as reused except the change back to self (if any?).

Also assert_balances should be ~all used, no?

Reply:
There are 11 outputs of 1 BTC at the address and 10 are summed up in a group (`OUTPUT_GROUP_MAX_ENTRIES`) which is then being used for the inputs. So only one output of 1 BTC is left at the address, 5 BTC have been sent to `ret_addr` and 5 BTC are in a change address (`total_sum=6`).

Reply:
You're right. Thanks for explaining.